### ISSUE TITLE
Feature Request: Highlighting and Comments?

### ISSUE THREAD

**Issue by Adrianzo** [rstudio/bookdown#272]
I think it would be great if rendered HTML gave readers the ability to highlight and/or comment at the side of the text. Just like in medium.com, gitbook.com or, better, openreviewtoolkit.org

**Comment by benmarwick** [comment]
You can add these features with [hypothes.is](https://hypothes.is/), this is what [openreviewtoolkit.org](https://openreviewtoolkit.org) is using.

I have implemented hypothes.is, and several other openreviewtoolkit.org features in bookdown here:  https://github.com/benmarwick/bookdown-ort

**Comment by yihui** [comment]
Yes, hypothes.is can be used for highlighting text and leaving comments. The problem is that its style does not work very well with the gitbook style -- its toolbar has some overlap with gitbook's toolbar:

![image](https://cloud.githubusercontent.com/assets/163582/21240805/7f7e375c-c2d2-11e6-8fa4-45f00edd43bc.png)

Besides this cosmetic issue, the more serious issue is that hypothes.is cannot send notifications for all comments on a website. You can only subscribe to a single page. This is pretty bad, since a book often has multiple pages, and it will be frustrating to subscribe to all of them one by one. I have contacted their team, but I don't think this issue can be resolved any time soon. Perhaps some Python programmers can help them.

For normal comments, you can consider Disqus: https://bookdown.org/yihui/bookdown/collaboration.html I don't like Disqus, either, mainly because it has no interest in supporting Markdown. It is so clunky to have to type HTML tags in the comments. But there is not a better alternative at the moment.
